Detectors/TRD/calibration/include/TRDCalibration/DCSProcessor.h

```diff
@@ -117,24 +139,52 @@ class DCSProcessor
   CcdbObjectInfo mCcdbCurrentsDPsInfo;
   CcdbObjectInfo mCcdbEnvDPsInfo;
   CcdbObjectInfo mCcdbRunDPsInfo;
+  // LB: new DPs for Fed and Cavern
+  CcdbObjectInfo mCcdbFedChamberStatusDPsInfo;
+  CcdbObjectInfo mCcdbFedCFGtagDPsInfo;
+  CcdbObjectInfo mCcdbFedEnvTempDPsInfo;
+  CcdbObjectInfo mCcdbCavernDPsInfo;
+
   TFType mGasStartTS;      ///< the time stamp of the first TF which was processesd for the current GAS CCDB object
   TFType mVoltagesStartTS; ///< the time stamp of the first TF which was processesd for the current voltages CCDB object
   TFType mCurrentsStartTS; ///< the time stamp of the first TF which was processesd for the current voltages CCDB object
   TFType mEnvStartTS;
   TFType mRunStartTS;
   TFType mRunEndTS;
+  // LB: new DPs for Fed and Cavern
+  TFType mFedChamberStatusStartTS;
+  TFType mFedCFGtagStartTS;
+  TFType mFedEnvTempStartTS;
+  TFType mCavernStartTS;
   TFType mCurrentTS{0}; ///< the time stamp of the TF currently being processed
   bool mGasStartTSset{false};
   bool mVoltagesStartTSSet{false};
   bool mCurrentsStartTSSet{false};
   bool mEnvStartTSSet{false};
   bool mRunStartTSSet{false};
+  // LB: new DPs for Fed and Cavern
+  bool mFedChamberStatusStartTSSet{false};
+  bool mFedCFGtagStartTSSet{false};
+  bool mFedEnvTempStartTSSet{false};
+  bool mCavernStartTSSet{false};
   std::bitset<constants::MAXCHAMBER> mVoltageSet{};
   bool mShouldUpdateVoltages{false};
   bool mShouldUpdateRun{false};
+  // LB: FedChamberStatus and FedCFGtag logic
+  bool mFedChamberStatusCompleteDPs{false};
+  bool mFedCFGtagCompleteDPs{false};
+  bool mNewRunForFedChamberStatusUpdate{false};
+  bool mNewRunForFedCFGtagUpdate{false};
```

Comment:
why do you need to flags here? Shouldn't there only be one 'mNewRun' needed?

Reply:
Alright, this is connected to the other conversation and I'll answer both here:

We need 4 variables in total, two for each ChamberStatus and CFGtag. This is due to the fact that the objects get updated to CCDB separately and the flags are cleaned after the update:

- The flags considering if the DPs are complete must be two (for each), because one object can be incomplete but not the other (at least in simulated data).
- The "run flags" need to also be separated because once one object is updated, it would "block" the updated of the following. For example, if the flag `mNewRun` was used for both: a new run begins -> `mNewRun = true`, ChamberStatus is updated -> `mNewRun` = false (to just upload first entry) -> CFGtag cannot be updated.

This flag also cannot be reset after a run update in case a new run number starts -> `mNewRun = true`, ChamberStatus is incomplete, run is updated -> `mNewRun = false`, complete ChamberStatus is sent but now it can't be updated.

I changed the name of variables to make this clearer, see commit https://github.com/AliceO2Group/AliceO2/pull/12079/commits/c1e9bcaa8af2752225fb1fd1165e3715813d0243.